src/Parser.php

```diff
@@ -216,20 +216,28 @@ function ($e) {
     }
 
     /**
-     * To avoid splitting on scoped package-names, every but the last @ are considered
-     * package name.
-     *
      * @param string $versionString
-     *
-     * @return string[]
+     * @return array{string, string}
+     * @throws ParserException
      */
     public static function splitVersionString($versionString)
     {
-         $parts = explode('@', $versionString);
-         $version = array_pop($parts);
-         return [
-             implode('@', $parts),
-             $version
-         ];
+        if (strpos($versionString, '@') === 0) {
```

Comment:
I preferred your original suggestion with the `strpos` with offset 1 – it either skips the namespace @ or it should work for single-letter package names with a result of 0 and was less noise imo.